db/version_set.cc

```diff
@@ -1481,6 +1481,32 @@ uint64_t Version::GetSstFilesSize() {
   return sst_files_size;
 }
 
+void Version::GetFilesViolatingTtl(std::vector<std::string>& files) {
+  int64_t temp_current_time = 0;
+  auto status = cfd_->ioptions()->env->GetCurrentTime(&temp_current_time);
+  if (!status.ok()) {
+    return;
+  }
+  const uint64_t current_time = static_cast<uint64_t>(temp_current_time);
+  for (int level = 0; level < storage_info_.num_levels_; level++) {
+    for (FileMetaData* meta : storage_info_.LevelFiles(level)) {
+      uint64_t file_creation_time =
+          meta->fd.table_reader->GetTableProperties()->file_creation_time;
+      if (file_creation_time == 0) continue;
+      uint64_t periodic_compaction_seconds =
+          cfd_->initial_cf_options().periodic_compaction_seconds;
+      uint64_t ttl = cfd_->initial_cf_options().ttl;
+      uint64_t time_limit = ttl > 0 ? ttl : periodic_compaction_seconds;
```

Comment:
Instead of inferring from TTL and compaction seconds, it might be a good idea to just return the oldest timestamp and the caller can determine what to do with it.